bc/core/utils/legal_provider.py

```diff
@@ -0,0 +1,331 @@
+# This is a simplified version of `LegalProvider` in CourtListener
+
+import random
+
+from faker import Faker
+from faker.providers import BaseProvider
+
+_faker = Faker()
+
+
+class LegalProvider(BaseProvider):
+    """
+    Generates fake legal-like citations, court names, and docket numbers.
+
+    This does not create all possible citation forms; it only creates
+    simple citations.
+    This uses a combination of fake & farcical words and
+    real or real-like words.
+    The list of reporters used is quite small.
+
+    As with other Faker providers, you can use this in code by doing:
+    # fmt: off
+    ```python
+        from faker import Faker
+        from bc.core.utils.legal_provider import LegalProvider
+
+        fake = Faker()
+        fake.add_provider(LegalProvider)
+    ```
+    # fmt: on
+    Now the methods in LegalProvider are available to fake (the
+    Faker instance you created).
+    Ex:
+      `fake.court_name()`
+      `fake.docket_number()`
+      `fake.citation()`
+      etc.
+
+    Because the methods are static, you can also use LegalProvider by
+    itself (without Faker). Ex:
+    # fmt: off
+    ```python
+        from bc.core.utils.legal_provider import LegalProvider
+
+        some_case_name = LegalProvider.case_name()
+        some_court = LegalProvider.court_name()
+    ```
+    # fmt: on
+    """
+
+    # Number of parties to create if creating a full (long) list of parties.
+    NUM_PARTIES = 5
+
+    @staticmethod
+    def citation_with_case() -> str:
+        """
+        Make a randomly generated citation that has a randomly generated
+        case name and a randomly generated citation.
+        Ex: `LegalProvider.citation_with_case()`
+            returns "Hooper, Gardner and Perry v. Bond Group, 579 L. Ed. 102"
+
+        :returns: the fake citation
+        :rtype: str
+        """
+        return f"{LegalProvider.case_name()}, {LegalProvider.citation()}"
+
+    @staticmethod
+    def citation() -> str:
```

Comment:
We don’t use citations in the bot. Is there a reason you’ve added it? If not, we should delete this method and the one that makes citations with case names.

Reply:
Definitely can do that.

I included them because they were in the CL version, and with an eye towards having this available as something that can be used for both Big Cases and CL.
But they can obv be added later if/when that happens.

I'll delete them. (They can always be re-added by looking at a previous commit.)

Reply:
Thanks. I think if we/you want to do a little faker package for legal data, they'd make a ton of sense there. We're sort of philosophically (?) opposed to adding unused code though, under the YAGNI banner. I'm not sure it's the right thing, but it's *our* think, anyway...